src/shared/tpm2-util.c

```diff
@@ -1487,6 +1487,73 @@ static int tpm2_get_policy_digest(
         return 0;
 }
 
+static int tpm2_create(
+                Tpm2Context *c,
+                const Tpm2Handle *parent,
+                const Tpm2Handle *session,
+                const TPMT_PUBLIC *template,
+                const TPMS_SENSITIVE_CREATE *sensitive,
+                TPM2B_PUBLIC **ret_public,
+                TPM2B_PRIVATE **ret_private) {
+
+        usec_t ts;
+        TSS2_RC rc;
+
+        assert(c);
+        assert(template);
+
+        log_debug("Creating object on TPM.");
+
+        ts = now(CLOCK_MONOTONIC);
+
+        TPM2B_PUBLIC tpm2b_public = {
+                .size = sizeof(*template),
+                .publicArea = *template,
+        };
+
+        /* Zero the unique area. */
+        zero(tpm2b_public.publicArea.unique);
+        tpm2b_public.size -= sizeof(tpm2b_public.publicArea.unique);
```

Comment:
why not initialize this above alreay, when you declare the function, i.e.:

```c
        TPM2B_PUBLIC tpm2b_public = {
                .size = sizeof(*template) - sizeof_field(TPMT_PUBLIC, unique),
                .publicArea = *template,
        };
```

Reply:
ack, updated